src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java

```diff
@@ -71,18 +69,13 @@ public static SubsetParameters parse(String subset) {
 
         //Validation
         if (timeRange != null) {
-            matcher = timePattern.matcher(subset);
-
-            while (matcher.find()) {
-                timeCount++;
-            }
-
-            if (timeCount != 2) {
-                throw new GoGoDuckException(String.format("Invalid time format for subset: %s Valid time format example: TIME,2009-01-01T00:00:00.000Z,2009-12-25T23:04:00.000Z", subset));
+            try {
+                CalendarDate startTime = CalendarDate.parseISOformat("gregorian", timeRange.start);
+                CalendarDate endTime = CalendarDate.parseISOformat("gregorian", timeRange.end);
+                calendarDateRange = CalendarDateRange.of(startTime, endTime);
+            } catch (IllegalArgumentException e) {
+                throw new GoGoDuckException(String.format("Invalid time format for subset: %s Valid time format example: TIME,2009-01-01T00:00:00.000Z,2009-12-25T23:04:00.000Z error: %s", subset, e.getMessage()));
             }
-            CalendarDate startTime = CalendarDate.parseISOformat("gregorian", timeRange.start);
-            CalendarDate endTime = CalendarDate.parseISOformat("gregorian", timeRange.end);
-            calendarDateRange = CalendarDateRange.of(startTime, endTime);
         }
 
         matcher = latLonPattern.matcher(subset);
```

Comment:
We've already proved that it matches this pattern by this point - why not remove this altogether like we have for time